adapters/repos/db/vector/hnsw/commit_logger.go

```diff
@@ -292,8 +294,23 @@ func (l *hnswCommitLogger) Reset() error {
 }
 
 func (l *hnswCommitLogger) StartLogging() {
-	// switch log
-	go func() {
+	// switch log job
+	cancelSwitchLog := l.startSwitchLogs()
+	// condense old logs job
+	cancelCondenseLogs := l.startCondenseLogs()
+	// cancel maintenance jobs on request
+	go func(cancel ...chan bool) {
```

Comment:
Do we ever send anything other than `true` on this `bool` channel? Could we make this a `chan struct{}` instead? Not meant as an optimization, more to illustrate that there is only a single value that can be sent over this channel.

Reply:
I can change this to `chan struct{}`, I would then change this in all of the places where I'm using `cancel` channel, so that it would be defined the same way everywhere